near-contract-standards/src/access/mod.rs

```diff
@@ -1,129 +1,279 @@
-            return false;
-        }
-        *self.roles.get(role).unwrap().members.get(account).unwrap_or(&false)
+    /// Returns `true` if `account` has been granted `role`.  
+    /// Otherwise, and on [missing `role`](Self::roles) or [missing `account`](RoleData::members), returns `false`.
+    pub fn has_role(&self, role: &RoleId, account: &AccountId) -> bool {
+        self.roles
+            .get(role)
+            .and_then(|role_data| role_data.members.get(account).cloned())
+            .unwrap_or(false)
     }
 
-    pub fn check_role(&self, role: &[u8; 32], account: &AccountId) {
+    /// Has no effect if `account` [has been granted](RoleData::members) `role`.  
+    /// Otherwise, and on [missing `role`](Self::roles) or [missing `account`](RoleData::members),
+    /// panics with a standard message.
+    ///
+    /// TODO: check this:  
+    /// The format of the panic message is given by the following regular expression:  
+    /// `/^AccessControl: account (0x[0-9a-f]{40}) is missing role (0x[0-9a-f]{64})$/`
+    ///
+    /// Uses [`Self::has_role()`] internally.
+    pub fn check_role(&self, role: &RoleId, account: &AccountId) {
         if !self.has_role(role, account) {
             env::panic_str(
                 format!("AccessControl: account {} is missing role {:?}", *account, *role).as_str(),
             )
         }
     }
 
-    pub fn only_role(&self, role: &[u8; 32]) {
+    /// Has no effect if the [`predecessor account`](env::predecessor_account_id())
+    /// has a specific `role`.  
+    /// Otherwise, and on [missing `role`](Self::roles) or [missing `account`](RoleData::members),
+    /// panics with a standard message including the required role.
+    ///
+    /// TODO: check this:  
+    /// The format of the panic message is given by the following regular expression:  
+    /// `/^AccessControl: account (0x[0-9a-f]{40}) is missing role (0x[0-9a-f]{64})$/`
+    ///
+    /// Uses [`Self::check_role()`] internally.
+    pub fn only_role(&self, role: &RoleId) {
         self.check_role(role, &env::predecessor_account_id());
     }
 
-    pub fn get_role_admin(&self, role: &[u8; 32]) -> [u8; 32] {
-        if !self.roles.contains_key(role) {
-            return self.default_admin_role.clone();
-        }
-        self.roles.get(role).unwrap().admin_role.clone()
+    /// Returns the [`admin role`](RoleData::admin_role) that controls `role`.  
+    /// Othewise on [missing `role`](Self::roles), returns [`Self::default_admin_role`].
+    ///
+    /// See also [`Self::grant_role()`] and [`Self::revoke_role()`].  
+    /// See [`Self::set_role_admin()`] to change a role's [admin role](RoleData::admin_role).
+    pub fn get_role_admin(&self, role: &RoleId) -> RoleId {
+        self.roles
+            .get(role)
+            .map(|role_data| role_data.admin_role)
+            .unwrap_or(self.default_admin_role)
     }
 
-    fn grant_role_internal(&mut self, role: [u8; 32], account: AccountId) {
```

Comment:
I renamed `*_internal` into `internal_*`, as this appears to be the pattern that is used elsewhere